tests/testAfterburners.py

```diff
@@ -0,0 +1,181 @@
+import numpy as np
+
+import unittest
+import lsst.utils.tests as utilsTests
+
+import lsst.meas.base.afterburner as AB
+import lsst.afw.table as afwTable
+from lsst.meas.base.pluginsBase import BasePlugin
+from lsst.meas.base.pluginRegistry import register
+from lsst.meas.base.baseLib import MeasurementError
+
+@register("afterburnerFail")
+class FailAb(AB.AfterburnerPlugin):
+    '''
+    Afterburner plugin which is guaranteed to fail, testing the failure framework
+    '''
+    @classmethod
+    def getExecutionOrder(cls):
+        return cls.DEFAULT_AFTERBURNER
+
+    def __init__(self, config, name, schema, metadata):
+        AB.AfterburnerPlugin.__init__(self, config, name, schema, metadata)
+        self.failKey = schema.addField(name+"_fail", type="Flag", doc="Failure test")
+
+    def burn(self, measRecord):
+        raise MeasurementError("Supposed to fail", 0)
+
+    def fail(self, measRecord, error=None):
+        measRecord.set(self.failKey, True)
+
+@register("singleRecordAfterburner")
+class SingleRecordAb(AB.AfterburnerPlugin):
+    '''
+    Afterburner plugin which works in single mode. It takes a single record, reads a value, squares it, and
+    writes out the results to the record
+    '''
+    @classmethod
+    def getExecutionOrder(cls):
+        return cls.DEFAULT_AFTERBURNER
+
+    def __init__(self, config, name, schema, metadata):
+        AB.AfterburnerPlugin.__init__(self, config, name, schema, metadata)
+        self.failKey = schema.addField(name+"_fail", type="Flag", doc="Failure flag")
+        self.squareKey = schema.addField(name+"_square", type="D", doc="Square of input catalog")
+
+    def burn(self, measRecord):
+        value = measRecord.get("start")
+        measRecord.set(self.squareKey, value**2)
+
+    def fail(self, measRecord, error=None):
+        measRecord.set(self.failKey, True)
+
+@register("multiRecordAfterburner")
+class MultiRecordAb(AB.AfterburnerPlugin):
+    '''
+    Afterburner plugin to test the framework in multimode. This plugin takes the whole source catalog at
+    onece, and loops over the catalog internally. The algorithm simply reads a value, cubes it, and writes the
+    results out to the table
+    '''
+    @classmethod
+    def getExecutionOrder(cls):
+        return cls.DEFAULT_AFTERBURNER
+
+    @property
+    def PlugType(self):
```

Comment:
That is a good point, it was an artifact of testing that I missed changing. It is updated to a class variable called plugType.